ESP_ATMod/ESP_ATMod.ino

```diff
@@ -324,30 +437,42 @@ void loop()
 					if (PemCertificatePos > MAX_PEM_CERT_LENGTH - 1)
 					{
 						gsCertLoading = false;
-						Serial.printf_P(MSG_ERROR);  // Invalid data
+						Serial.printf_P(MSG_ERROR); // Invalid data
 					}
 				}
 			}
 			else if (c < ' ')
-			{}
-			else  // illegal character in certificate
+			{
+			}
+			else // illegal character in certificate
 			{
 				gsCertLoading = false;
-				Serial.printf_P(MSG_ERROR);  // Invalid data
+				Serial.printf_P(MSG_ERROR); // Invalid data
+			}
+
+			if (!gsCertLoading)
+			{
+				// Read everything left before continuing
+				while (Serial.available() > 0)
+				{
+					// Read the incoming byte
+					int incomingByte = Serial.read();

```

Comment:
No need to declare an unused variable. `Serial.read();` is sufficient.

Reply:
True indeed, used this earlier in my test but didn’t remove the variable. Resolved now.